uqcsbot/scripts/welcome.py

```diff
@@ -0,0 +1,39 @@
+# Welcomes new users to UQCS Slack and check for member milestones
+from uqcsbot import bot
+import asyncio
+
+MEMBER_MILESTONE = 50;  # Number of members between posting a celebration
+MESSAGE_PAUSE = 2500 / 1000;   # Number of seconds between sending bot messages
+WELCOME_MESSAGES = [    # Welcome messages sent to new members
+    "Hey there! Welcome to the UQCS slack!",
+    "This is the first time I've seen you, so you're probably new here",
+    "I'm UQCSbot, your friendly (open source) robot helper",
+    "We've got a bunch of generic channels (e.g. #banter, #games, #projects) along with many subject-specific ones",
+    "Your friendly admins are @csa, @rob, @mb, @trm, @mitch, @guthers, and @artemis",
+    "Type \"help\" here, or \"!help\" anywhere else to find out what I can do!",
+    "and again, welcome :)"
+];
+
+@bot.on("member_joined_channel")
+async def welcome(evt: dict):
+    chan = bot.channels.get(evt.get('channel'))
+    if chan is None or chan.name != "announcements":
+        return
+    announcements = chan
+    general = bot.channels.get("general")
+    user_info = await bot.run_async(bot.api.users.info, user=evt.get("user"))
+    display_name = user_info.get("user", {}).get("profile", {}).get("display_name")
```

Comment:
If the display_name ends up being null here, we should not try and post a welcome message I think saying `"Welcome, None"`